guides/micronaut-cache-hazelcast/micronaut-cache-hazelcast.adoc

```diff
@@ -0,0 +1,166 @@
+<4> Emulate an expensive operation by sleeping for several seconds.
+<5> The return value is cached with name `headlines` for the supplied `month`. The method invocation is never skipped even if the cache `headlines` for the supplied `month` already exists.
+<6> Method invocation causes the invalidation of the cache `headlines` for the supplied `month`.
+
+TIP: If you don't annotate the class with `@CacheConfig`, specify the cache name in the cache annotations. E.g. `@Cacheable(value = "headlines", parameters = {"month"})`
+
+=== Test the Cache
+
+We use https://testcontainers.org[Testcontainers] to test the application.
+
+Create a class to ease the start-up and shutdown of Hazelcast with TestContainers.
+
+test:HazelcastUtils[]
+
+We can verify that the cache works as expected:
+
+test:NewsServiceTest[]
+
+:exclude-for-languages:groovy
+callout:test-method-order[1]
+callout:micronaut-test-start-application-false[2]
+callout:test-instance-per-class[3]
+callout:test-property-provider[4]
+<5> Inject `NewsService` bean.
+callout:timeout[6]
+callout:order[7]
+:exclude-for-languages:
+
+:exclude-for-languages:java,kotlin
+callout:stepwise[1]
+callout:micronaut-test-start-application-false[2]
+callout:test-property-provider[3]
+<4> Inject `NewsService` bean.
+callout:timeout[5]
+:exclude-for-languages:
+
+=== Controller
+
+Create a controller which engages the previous service:
+
+source:News[]
+
+callout:introspected[1]
+
+source:NewsController[]
+
+callout:controller[number=1,arg0=/]
+
+Add a test. We call the endpoint twice and verify with the `@Timeout` annotation that the cache is being used.
+
+test:NewsControllerTest[]
+
+:exclude-for-languages:groovy
+callout:test-method-order[1]
+callout:micronaut-test[2]
+callout:test-instance-per-class[3]
+callout:test-property-provider[4]
+callout:http-client[5]
+callout:timeout[6]
+callout:order[7]
+:exclude-for-languages:
+
+:exclude-for-languages:java,kotlin
+callout:stepwise[1]
+callout:micronaut-test[2]
+callout:test-property-provider[3]
+callout:http-client[4]
+callout:timeout[5]
+:exclude-for-languages:
+
+common:testApp.adoc[]
+
+=== Configure Hazelcast 
+
+The minimal configuration to use Hazelcast is to simply declare `hazelcast:` with a network configuration for addresses of the Hazelcast cluster:
+
+resource:application.yml[tag=hazelcast]
+
+Replace `192.168.4.67` with your host machine address. See following section.
+
+== Running Hazelcast via Docker 
+
+Run https://hub.docker.com/r/hazelcast/hazelcast[Hazelcast via Docker]
+
+Create a Hazelcast cluster with two Docker containers and access it from the client application.
+
+[source, bash]
+----
+docker run -e HZ_NETWORK_PUBLICADDRESS=192.168.4.67:5701 -p 5701:5701 hazelcast/hazelcast:4.2.1
+----
+[source, bash]
+----
+docker run -e HZ_NETWORK_PUBLICADDRESS=192.168.4.67:5702 -p 5702:5701 hazelcast/hazelcast:4.2.1
+----
+
+Replace `192.168.4.67` with your host machine address. It is used for the Hazelcast communication.
```

Comment:
What does "it" refer to at the beginning of the second sentence? The address? Might be more clear if you replaced "it" with "this."

Reply:
Yes, it refers to the address.